Dungeon Loot Spawners Rework

ErhardSteinhauer opened the pull request:
## About the PR
A long-overdue rework of dungeon spawners. This update transitions from the old format to loot tables, which offer much greater control over spawn probabilities. The PR will remain in draft for a while, as completing it will take some time. The plan is to review and update all `dungeon_item_*.yml` files in `Resources/Prototypes/_NF/Entities/Markers/Spawners/Random`.

**Progress**:
- [x] Corpses.
- [x] General.
- [x] Engineering.
- [x] Hydroponics.
- [x] Kitchen.
- [x] Medical.
- [x] Mercenary.
- [x] Research.
- [x] Supply.
- [x] Weapons.

## Why / Balance
Maintenance

## Technical details
Too much yml

The goal is to further atomize loot by type and theme—beyond what was done in the original loot spawner PR. Additionally, loot tables and spawners will be split into separate files for better organization.

Spawn probabilities have been adjusted so that rarer items appear slightly more frequently.

Two distinct desired distributions of spawn probabilities (that are only partially implemented):
1. Quasilinear. Used for flavor / clutter category of items.
  - T1 - "Common" - 0.4
  - T2 - "Uncommon" - 0.3
  - T3 - "Rare" - 0.15
  - T4 - "Epic" - 0.1
  - T5 - "Legendary" - 0.05. Either really good items or something utterly undesirable.
  - "Unique" - 0.01. Not always present.
2. Skewed. Used for more valuable categories of items.
  - T1 - "Common" - 0.15
  - T2 - "Uncommon" - 0.3
  - T3 - "Rare" - 0.35
  - T4 - "Epic" - 0.15
  - T5 - "Legendary" - 0.05. Either really good items or something utterly undesirable.
  - "Unique" - 0.01. Not always present.

## How to test
1. Primarily peruse the tables.
3. Can spam expeditions.

## Media
<!-- Attach media if the PR makes ingame changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [x] I have read [CONTRIBUTING.md](https://github.com/new-frontiers-14/frontier-station-14/blob/master/CONTRIBUTING.md) and and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.
- [x] I have reviewed the [Ship Submission Guidelines](https://frontierstation.wiki.gg/wiki/Ship_Submission_Guidelines) if relevant.
- [x] I confirm that the content in this PR is my own work, and/or is properly attributed to the original author(s).
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them. -->

**Changelog**
:cl:
- tweak: Adjusted loot pool and spawn probabilities for expeditions: valuables are slightly more prevalent, more useful items in the pool.

blackknight954 reviewed (commented):
Tested and seemed to work fine, apart from spawning... random machine boards and flatpacks on the bar at exped lodge?

dustylens commented on Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/Items/general_tables.yml:
```diff
@@ -0,0 +1,869 @@
+    - id: NFCrateFunSyndicateSegway
+      weight: 0.1
+    - id: CrateFunScurret
+      weight: 0.05
```
Did we want Scurrets in the vehicle crate spawn table?

ErhardSteinhauer replied:
It comes with a saddle. Obviously it is a vehicle then.

dustylens commented:
Ok, got a little time after work to sink some time into testing this PR. 

I found that my experience was essentially 1:1 with Hive's. So I don't believe I have much to contribute there beyond providing my own experiences.

I had enough time around cooking soup to solo run two VG roids (clearing out one of the various structure complexes on each) and I walked away with a full set of service, salvage, engineering techfabs in addition a half dozen various exosuit/shredder/what have you lathes that I eventually stopped picking up. In addition to that I walked out with over two stacks of bluespace manipulators, a stack of bins and a stack and a half of bluespace capacitors. Nevermind how many T3 and below parts. Should have taken a screenshot but I think I had a little over three stacks of super manipulators.

Also a full kitchen setup, but that was a bit more of a luck of the draw with the room generation. Both were salvager VGroids.

I feel that non departmental lathes and bluespace parts are some of our most substantial carrots, beyond fun and interesting guns. Part of me feels that it might not be unwise to leave those out of approachable lootpools and leave them to deliberate mapping (outside of specific very very deep asteroid spawn pools, such as we enjoy with the brassteroids and similar future content.

ErhardSteinhauer commented on Resources/Prototypes/_NF/Entities/Markers/Spawners/Random/Items/engineering_tables.yml:
```diff
@@ -0,0 +1,783 @@
+      weight: 0.08
+      children:
+      - id: RadiationCollectorFullTankFlatpack
+      - id: TeslaCoilFlatpack
```
I believe the issue in question doesn’t actually exist, for several reasons:
1. Players generally don’t know which items are included (or excluded) in loot pools.
2. No flatpack spawners are specifically mapped on any dungeon templates, meaning all flatpacks are drawn from compound spawn tables which further dilutes individual spawn chances. In practice, this likely results in roughly one randomly spawned radiation collector flatpack per shift, or even fewer.
3. Engineers don’t run expeditions or VGroids. Even if they were aware that radiation collectors could be obtained this way, they’d have to put out a bounty.
4. Expedition and VGroid runners rarely loot complex machinery if it's not a ready to go pack (because you can't target farm specific items) which means that there is little incentive for a salvager to loot radiation collector: it is not as valuable (talking about currency value) as other items laying on the floor next to it, and it is a part of complex machinery, having one radcollector does you no good - you need a potent radiation source.
5. If there is a bounty for a radiation collector present (first big IF), exped/vgroid runner reads it (another big IF) and then actually finds one (another even bigger IF) and then actually delivers the bounty to an engineer (the fourth IF) only then the engineers will obtain one more radiation collector. That's 17% increase in number of radiation collectors available to them for one shift (part of a shift at any rate).

I also don’t feel strongly about it, so if more requests come in to adjust things (especially changes with a greater perceived impact) I’ll simply comment out the radcollector along with the other tweaks.

ErhardSteinhauer commented:
Went through the tables and adjusted the distributions here and there:
- for clutter tables: made flatpacks, circuit boards and parts less common;
- for parts tables: changed the distribution so it no longer favors the bluespace parts - heavily skewed towards advanced parts, but super parts should be somewhat common, BS should be much rarer);
- for machines tables: made so the advanced ones (hyper-whatever) are less common, department specific ones are still fairly easy to come by;
- commented out radcollectors.

blackknight954 commented:
Hyper convection lathes are barely used on Frontier. I'd imagine they'd be passed up to be completely honest

>commented out radcollectors.

Something come up in testing that made em more common or something?

ErhardSteinhauer commented:
> I am concerned that the newer buffed loot tables are being used for bundles as well.

In the [follow up PR that adjusts NPC loot](https://github.com/new-frontiers-14/frontier-station-14/pull/4064) I'm changing that:
- Parts and Flatpacks spawners are reserved for bosses only.
- For the most part NPCs will drop either mats, cash, weapons (unmodified by weapon randomization) or thematic vendor fluff.

ErhardSteinhauer commented:
> Hyper convection lathes are barely used on Frontier. I'd imagine they'd be passed up to be completely honest

Given that the hyper-whatever lathes and their non-hyper-whatever variants are not so great this change wont move the needle

> Something come up in testing that made em more common or something?

Not really. I just decided that if I don't care I'll roll with the voiced opinion of someone who cares about it.

Mackazonia98 commented:
Having the _possibility_ of getting more rad collectors is a nice theoretical thing for PMs to hope for. I share Tibels experience with never having seen a single collector or coil be delivered to edison. I've witnessed mercs find them once or twice, but the odds of them bothering to collect it, read the bounty, and actually make the trip is low. I don't see an issue with rad collectors remaining something you can stumble across.

pogcat09 commented:
yeah i agree we should keep rad collectors as possible loot

dustylens reviewed (approved):
Did a VG roid run through and the general loot expectations seem much more aligned with expectation. A good batch of advanced and some super parts with a few higher end rare items.

I even picked up an RPG. My lucky day. 

On the whole I think this is a good spot. I would appreciate a second pair of eyes on the yml and the practical result in game.

blackknight954 commented:
Tested and the adjustments do work as intended, yes.

Might be a bit overtuned, might not, we'll have to see how it goes live. For what it's worth it seems fine to me though.
